src/lib/services/AuthenticationService/AuthenticationService.ts

```diff
@@ -12,13 +12,24 @@ async function sendPin(body: SendPinBody): Promise<void> {
   localStorage.setItem(OTP_AUTH_KEY, response.otp_auth_key);
 }
 
-function verify(body: VerifyBody): Promise<VerifyResponse> {
-  return client.post("auth/verify", body);
+async function authenticate(pin: string): Promise<void> {
+  const body: VerifyBody = {
+    otp_auth_key: localStorage.getItem(OTP_AUTH_KEY) ?? "",
+    pin,
+  };
+  const response: VerifyResponse = await client.post("auth/verify", body);
+  localStorage.setItem(GC_API_TOKEN, response.jwt_token);
+  localStorage.removeItem(OTP_AUTH_KEY);
```

Comment:
Thinking out loud. Since the purpose of storing `OTP_AUTH_KEY` is entirely to handle refresh. It might actually belong in the Login pages.

Perspective: If you move this service out of the project, people would wonder why you're storing it